File: app/src/main/java/me/tagavari/airmessage/util/TrackableRequest.java
package me.tagavari.airmessage.util;

import me.tagavari.airmessage.enums.TrackableRequestCategory;

/**
 * Helper interface for finding requests based on a category and key value
 * @param <T> The type of the value
 */
public interface TrackableRequest<T> {
	@TrackableRequestCategory
	int getCategory();
	T getValue();
}

File: common/me/tagavari/airmessage/util/TrackableRequest.kt
package me.tagavari.airmessage.util

import me.tagavari.airmessage.enums.TrackableRequestCategory

/**
 * Helper interface for finding requests based on a category and key value
 * @param <T> The type of the value
*/
interface TrackableRequest<T> {
	@get:TrackableRequestCategory val category: Int
	val value: T
}